sdk/core/Azure.Core/tests/ResponseTests.cs

```diff
@@ -87,6 +90,32 @@ public void ToStringsFormatsStatusAndMessageForNoBodyResponse()
             Assert.AreEqual("Status: 200, Service returned no content", response.ToString());
         }
 
+        [Test]
+        public void ContentPropertyGetsContent()
+        {
+            var response = new MockResponse(200);
+            Assert.AreEqual(0, response.Content.ToArray().Length);
+
+            var responseWithBody = new MockResponse(200);
+            responseWithBody.SetContent("body content");
+            Assert.AreEqual("body content", responseWithBody.Content.ToString());
+
+            // Ensure that the BinaryData is formed over the used portion of the memory stream, not the entire buffer.
```

Comment:
SHouldn't it be the opposite?

Reply:
Maybe I wrote the comment in a bad way.  My goal in this test is that we ensure the `BinaryData` that is retuned from `.Content` only contains data from the valid part of the stream (and not the view of the entire underlying buffer).  In this test we create a MemoryStream with a backing buffer of 50 bytes, don't write anything to it and then confirm the Content returned is empty (and not 50 bytes of 0's).

Reply:
I'm worried about a different scenario:

We get a response, the client reads the content stream to deserialize it (https://github.com/Azure/azure-sdk-for-net/issues/17711) and doesn't rewind it back. What would the `Content` be in that case?

Reply:
> What would the Content be in that case?

Ahh - I understand.  I think that in all these cases we want `Content` to be the entire contents of the stream, correct?  Having it mean: "read from the current stream position to the end of the stream" feels wrong?

Reply:
>I think that in all these cases we want Content to be the entire contents of the stream, correct


Yep, I think so too.

Reply:
Added another test here to ensure we have the right behavior